### ISSUE TITLE
Windows: Cannot start server in new window because no terminal app was specified

### ISSUE THREAD

**Issue by feng-yu-healthbank** [react-native-community/cli#2219]
**Do you want to request a *feature* or report a *bug*?**

Report a *bug*

It's originally raised on StackOverflow https://stackoverflow.com/questions/77684266/react-native-error-cannot-start-server-in-new-window-because-no-terminal-app-wa/77688826

**What is the current behavior?**

When launching react-native metro on **Windows**, it gets "error Cannot start server in new window because no terminal app was specified".

**If the current behavior is a bug, please provide the steps to reproduce and a minimal repository on GitHub that we can `yarn install` and `yarn test`.**

It's really just a fresh react-native project using the standard init.

```powershell
npx react-native@latest init AwesomeProject && cd ./AwesomeProject && npm run android
```

If you don't like to run it yourself, I have created a repo for you. No customization whatsoever, just a fresh react-native project.

https://github.com/feng-yu-healthbank/AwesomeProject

Please note the error is specific to Windows.

**What is the expected behavior?**

The empty project launches into an Android emulator by `npm run android`.

**Please provide your exact Metro configuration and mention your Metro, node, yarn/npm version and operating system.**

It's a fresh official react-native sample project. No customization is applied.

Metro version
```
"metro": "^0.80.0"
```

metro config: 
```
const {getDefaultConfig, mergeConfig} = require('@react-native/metro-config');

/**
 * Metro configuration
 * https://facebook.github.io/metro/docs/configuration
 *
 * @type {import('metro-config').MetroConfig}
 */
const config = {};

module.exports = mergeConfig(getDefaultConfig(__dirname), config);
```

npx react-native info
```
System:
  OS: Windows 11 10.0.22621
  CPU: (24) x64 13th Gen Intel(R) Core(TM) i7-13700K
  Memory: 27.55 GB / 63.77 GB
Binaries:
  Node:
    version: 20.10.0
    path: C:\Program Files\nodejs\node.EXE
  Yarn:
    version: 1.22.19
    path: ~\AppData\Roaming\npm\yarn.CMD
  npm:
    version: 9.7.1
    path: C:\Program Files\nodejs\npm.CMD
  Watchman:
    version: 20231008.002904.0
    path: C:\ProgramData\chocolatey\bin\watchman.EXE
SDKs:
  Android SDK:
    Android NDK: 22.1.7171670
  Windows SDK:
    AllowAllTrustedApps: Disabled
    Versions:
      - 10.0.19041.0
IDEs:
  Android Studio: AI-231.9392.1.2311.11076708
  Visual Studio: Not Found
Languages:
  Java: 17.0.9
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.1
    wanted: 0.73.1
  react-native-windows: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```

![image](https://github.com/facebook/metro/assets/125230892/62629e9b-61e5-409d-9dfd-730bd24ea0dd)

**Comment by Navpan18** [comment]
Facing same issue

**Comment by Navpan18** [comment]
this problem is occuring from react native version 0.72.9

**Comment by coder-xiaomo** [comment]
Same here. I upgraded from _react-native@0.72_ to _react-native@0.73.1_, and then, the problem occurred.

I found a temporary solution that is to use `npm run start` or `npm start` instead of `npm run android` mentioned in [this stackoverflow page](https://stackoverflow.com/questions/77684266/react-native-error-cannot-start-server-in-new-window-because-no-terminal-app-wa)

**Comment by Navpan18** [comment]
App works fine till react native version 0.72.8 . Problem begins from 0.72.9

**Comment by Navpan18** [comment]
> Same here. I upgraded from _react-native@0.72_ to _react-native@0.73.1_, and then, the problem occurred.
> 
> I found a temporary solution that is to use `npm run start` or `npm start` instead of `npm run android` mentioned in [this stackoverflow page](https://stackoverflow.com/questions/77684266/react-native-error-cannot-start-server-in-new-window-because-no-terminal-app-wa)

Check if your app works fine in RN 0.72.8

**Comment by szymonrybczak** [comment]
Hey @Navpan18, @coder-xiaomo can you try passing via `--terminal` flag a path to executable file of your terminal? and say if that is working :)

**Comment by szymonrybczak** [comment]
> This works for me. However it only opens a new window and doesn't automatically run andoid, it acts as if you've run react-native start only

Interesting, can you please record a video which presents this behaviour?

**Comment by james-cohen** [comment]
Sure, I've recorded 2 GIFs below. When I run 'yarn android:test' it is the following npm script:

```
react-native run-android --active-arch-only --mode=devDebug
```

The first video, you can see that even after waiting for over a minute, the Android build doesn't start. You have to press a in the Metro terminal but then all of the parameters are lost

![metro1](https://github.com/react-native-community/cli/assets/61296803/f7ef6b00-2b3a-4735-916a-40e5a50ea186)

This second video, I run the same script with Metro already running and the Android build starts immediately

![metro2](https://github.com/react-native-community/cli/assets/61296803/743afb09-ee40-41ee-8491-987666d4e4cd)

**Comment by abbasbphppoets** [comment]
> Sure, I've recorded 2 GIFs below. When I run 'yarn android:test' it is the following npm script:
> 
> ```
> react-native run-android --active-arch-only --mode=devDebug
> ```
> 
> The first video, you can see that even after waiting for over a minute, the Android build doesn't start. You have to press a in the Metro terminal but then all of the parameters are lost
> 
> ![metro1](https://private-user-images.githubusercontent.com/61296803/292560026-f7ef6b00-2b3a-4735-916a-40e5a50ea186.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NzAyNjAsIm5iZiI6MTcwMzY2OTk2MCwicGF0aCI6Ii82MTI5NjgwMy8yOTI1NjAwMjYtZjdlZjZiMDAtMmIzYS00NzM1LTkxNmEtNDBlNWE1MGVhMTg2LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDA5MzkyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5ZjJmNDdlYWVhMGZlNTgxYjQ3NDEzYWUzNTFhMzNhYzZlOWZlMTczZmZhOWEwZTMxZDgyNDhjNGM5NTY4ZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4TH0AHLB8W6MTGDZJ2XBIAgPe9csAUzfF0ousNtd4tg) [ ![metro1](https://private-user-images.githubusercontent.com/61296803/292560026-f7ef6b00-2b3a-4735-916a-40e5a50ea186.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NzAyNjAsIm5iZiI6MTcwMzY2OTk2MCwicGF0aCI6Ii82MTI5NjgwMy8yOTI1NjAwMjYtZjdlZjZiMDAtMmIzYS00NzM1LTkxNmEtNDBlNWE1MGVhMTg2LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDA5MzkyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5ZjJmNDdlYWVhMGZlNTgxYjQ3NDEzYWUzNTFhMzNhYzZlOWZlMTczZmZhOWEwZTMxZDgyNDhjNGM5NTY4ZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4TH0AHLB8W6MTGDZJ2XBIAgPe9csAUzfF0ousNtd4tg) ](https://private-user-images.githubusercontent.com/61296803/292560026-f7ef6b00-2b3a-4735-916a-40e5a50ea186.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NzAyNjAsIm5iZiI6MTcwMzY2OTk2MCwicGF0aCI6Ii82MTI5NjgwMy8yOTI1NjAwMjYtZjdlZjZiMDAtMmIzYS00NzM1LTkxNmEtNDBlNWE1MGVhMTg2LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDA5MzkyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5ZjJmNDdlYWVhMGZlNTgxYjQ3NDEzYWUzNTFhMzNhYzZlOWZlMTczZmZhOWEwZTMxZDgyNDhjNGM5NTY4ZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4TH0AHLB8W6MTGDZJ2XBIAgPe9csAUzfF0ousNtd4tg) [ ](https://private-user-images.githubusercontent.com/61296803/292560026-f7ef6b00-2b3a-4735-916a-40e5a50ea186.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NzAyNjAsIm5iZiI6MTcwMzY2OTk2MCwicGF0aCI6Ii82MTI5NjgwMy8yOTI1NjAwMjYtZjdlZjZiMDAtMmIzYS00NzM1LTkxNmEtNDBlNWE1MGVhMTg2LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDA5MzkyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5ZjJmNDdlYWVhMGZlNTgxYjQ3NDEzYWUzNTFhMzNhYzZlOWZlMTczZmZhOWEwZTMxZDgyNDhjNGM5NTY4ZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4TH0AHLB8W6MTGDZJ2XBIAgPe9csAUzfF0ousNtd4tg)
> 
> This second video, I run the same script with Metro already running and the Android build starts immediately
> 
> ![metro2](https://private-user-images.githubusercontent.com/61296803/292560031-743afb09-ee40-41ee-8491-987666d4e4cd.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NzAyNjAsIm5iZiI6MTcwMzY2OTk2MCwicGF0aCI6Ii82MTI5NjgwMy8yOTI1NjAwMzEtNzQzYWZiMDktZWU0MC00MWVlLTg0OTEtOTg3NjY2ZDRlNGNkLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDA5MzkyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmZjZkMGMxNDY1OGE5NjVjZDRjOWQ3ZjM0ZDYyZTVlOTkyYTNlYmJhZGU4NzEwY2MxNjU1ZGE2MDViMzE3OGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OfvqcPUzgYcT6aa7GdlufcbgPQqkmdmn5KfLwVAsSWc) [ ![metro2](https://private-user-images.githubusercontent.com/61296803/292560031-743afb09-ee40-41ee-8491-987666d4e4cd.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NzAyNjAsIm5iZiI6MTcwMzY2OTk2MCwicGF0aCI6Ii82MTI5NjgwMy8yOTI1NjAwMzEtNzQzYWZiMDktZWU0MC00MWVlLTg0OTEtOTg3NjY2ZDRlNGNkLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDA5MzkyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmZjZkMGMxNDY1OGE5NjVjZDRjOWQ3ZjM0ZDYyZTVlOTkyYTNlYmJhZGU4NzEwY2MxNjU1ZGE2MDViMzE3OGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OfvqcPUzgYcT6aa7GdlufcbgPQqkmdmn5KfLwVAsSWc) ](https://private-user-images.githubusercontent.com/61296803/292560031-743afb09-ee40-41ee-8491-987666d4e4cd.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NzAyNjAsIm5iZiI6MTcwMzY2OTk2MCwicGF0aCI6Ii82MTI5NjgwMy8yOTI1NjAwMzEtNzQzYWZiMDktZWU0MC00MWVlLTg0OTEtOTg3NjY2ZDRlNGNkLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDA5MzkyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmZjZkMGMxNDY1OGE5NjVjZDRjOWQ3ZjM0ZDYyZTVlOTkyYTNlYmJhZGU4NzEwY2MxNjU1ZGE2MDViMzE3OGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OfvqcPUzgYcT6aa7GdlufcbgPQqkmdmn5KfLwVAsSWc) [ ](https://private-user-images.githubusercontent.com/61296803/292560031-743afb09-ee40-41ee-8491-987666d4e4cd.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NzAyNjAsIm5iZiI6MTcwMzY2OTk2MCwicGF0aCI6Ii82MTI5NjgwMy8yOTI1NjAwMzEtNzQzYWZiMDktZWU0MC00MWVlLTg0OTEtOTg3NjY2ZDRlNGNkLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDA5MzkyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmZjZkMGMxNDY1OGE5NjVjZDRjOWQ3ZjM0ZDYyZTVlOTkyYTNlYmJhZGU4NzEwY2MxNjU1ZGE2MDViMzE3OGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OfvqcPUzgYcT6aa7GdlufcbgPQqkmdmn5KfLwVAsSWc)

doesn't work

**Comment by WesleyMaciel2510** [comment]
Same issue, trying to solve it.

**Comment by dezudas** [comment]
+1 
$ react-native run-ios --simulator 'iPhone 15'
error Cannot start server in new window because no terminal app was specified.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by rishimagiawala** [comment]
> > Same issue, trying to solve it.
> 
> temporary is run script: npx react-native start and after that type mode a

This works for me, thx.

I ran "npx react-native start", then hit "a" when it gave me the option to

**Comment by Shreyas-29** [comment]
Try running the following command:😊

`npx react-native start`

**Comment by jslok** [comment]
Getting this bug on windows+android on 0.73.1. Works fine on 0.73.0. Not tested on mac/ios.

**Comment by machadinha** [comment]
When i run "npx react-native start", then hit "a" it gives me this 
![image](https://github.com/react-native-community/cli/assets/151222829/a87eb902-c432-4d49-84a6-cdffa9cf14b6)
what do i do

**Comment by yetki** [comment]
I fix my problem with 

npx react-native run-android --terminal powershell

however I want to use only  npx react-native run-android

**Comment by Shreyas-29** [comment]
> When i run "npx react-native start", then hit "a" it gives me this ![image](https://private-user-images.githubusercontent.com/151222829/293604346-a87eb902-c432-4d49-84a6-cdffa9cf14b6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyMDUwNTMsIm5iZiI6MTcwNDIwNDc1MywicGF0aCI6Ii8xNTEyMjI4MjkvMjkzNjA0MzQ2LWE4N2ViOTAyLWM0MzItNGQ0OS04NGE2LWNkZmZhOWNmMTRiNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwMlQxNDEyMzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03Nzc5YjdlYmM1ZTU1OGQ0YWU0YTExNjljODU5NDZhZDAwNmU2YTQ1YWFiMzZiZDcyZTk3NjI2Y2IxYzQ4YTEyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.7v3YClCKxH4wiBPih-TN5VdrX5rSGhbQ2H840Xkn90c) what do i do im in windowns

Yes. I am also getting that error. 
But don't know how to fix it. I am unable to run the app on neither physical device nor android studio.

**Comment by szymonrybczak** [comment]
@machadinha for you it is not working because you don't have `ANDROID_HOME` env variable provided - please re-read an error message and try setting in your system setting - also it may be helpful to run `npx react-native doctor` and make sure it is passing :)

**Comment by szymonrybczak** [comment]
> however I want to use only npx react-native run-android

you can also do `npx react-native run-android --terminal`

**Comment by machadinha** [comment]
> @machadinha @Shreyas-29 for you it is not working because you don't have `ANDROID_HOME` env variable provided - please re-read an error message and try setting in your system setting - also it may be helpful to run `npx react-native doctor` and make sure it is passing :)
when i run npx react-native doctor it gives me this what do I do
![image](https://github.com/react-native-community/cli/assets/151222829/d297860b-e1a6-43a5-b670-7de8bd9dfe93)

**Comment by machadinha** [comment]
> @machadinha @Shreyas-29 for you it is not working because you don't have `ANDROID_HOME` env variable provided - please re-read an error message and try setting in your system setting - also it may be helpful to run `npx react-native doctor` and make sure it is passing :)

How do I set the ANDROID_HOME

**Comment by machadinha** [comment]
> @machadinha @Shreyas-29 for you it is not working because you don't have `ANDROID_HOME` env variable provided - please re-read an error message and try setting in your system setting - also it may be helpful to run `npx react-native doctor` and make sure it is passing :)

Now when i run it gives me this 
![image](https://github.com/react-native-community/cli/assets/151222829/2e93e2b4-5c99-45b6-bac3-d0a6853137aa)
![image](https://github.com/react-native-community/cli/assets/151222829/ac8d3c25-5fee-4ca0-bbc5-c9eb9d8340e0)
![image](https://github.com/react-native-community/cli/assets/151222829/20b97c83-c723-4900-af23-4fe4a52aacae)
![image](https://github.com/react-native-community/cli/assets/151222829/f0079797-15fa-4c6c-9c0e-a07e313eaf29)
![image](https://github.com/react-native-community/cli/assets/151222829/76fa4f4c-3d1f-4030-a2f1-ae3845823456)
![image](https://github.com/react-native-community/cli/assets/151222829/9e356618-e386-4f35-b390-7865a405d878)

**Comment by machadinha** [comment]
> @machadinha @Shreyas-29 for you it is not working because you don't have `ANDROID_HOME` env variable provided - please re-read an error message and try setting in your system setting - also it may be helpful to run `npx react-native doctor` and make sure it is passing :)

Its resolver I thinks

**Comment by yetki** [comment]
> > however I want to use only npx react-native run-android
> 
> you can also do `npx react-native run-android --terminal`

PS FILE_PATH> npx react-native run-android --terminal
error: option '--terminal <string>' argument missing

It was asking string like node, powershell etc.

**Comment by yetki** [comment]
node_modules\@react-native-community\cli-platform-android\build\commands\runAndroid\index.js

```
  if (packager) {
    const {
      port: newPort,
      startPackager
    } = await (0, _cliTools().findDevServerPort)(port, config.root);
    if (startPackager) {
      await (0, _cliTools().startServerInNewWindow)(newPort, config.root, config.reactNativePath, args.terminal);
    }
  }
```


I remove the await in 
```
 if (startPackager) {
      (0, _cliTools().startServerInNewWindow)(newPort, config.root, config.reactNativePath, args.terminal);
    }```

and it worked. I saw it on @react-native-community

**Comment by linq8261** [comment]
> > > however I want to use only npx react-native run-android
> > 
> > 
> > you can also do `npx react-native run-android --terminal`
> 
> PS FILE_PATH> npx react-native run-android --terminal error: option '--terminal ' argument missing
> 
> It was asking string like node, powershell etc.

`npx react-native run-android --terminal C:\Windows\System32\cmd.exe`

**Comment by Francisco420** [comment]
Why when i run npx react-native@latest init PAP in the final it gives me this im in windowns
![image](https://github.com/react-native-community/cli/assets/156228945/a70cc165-e8f9-4f67-a0a4-5d5b347a7caa)

**Comment by XinrZhou** [comment]
`npm run start`
`npm start`
`npx react-native start`
Try the above commands~

**Comment by faraz-stack** [comment]
Simple Solution 
Run `npx react-native start`
and after when metro starts successfully and gives you Dev server ready options hit a to run on android and i to run on iOS.
it works for me.

**Comment by YoraiLevi** [comment]
why's this closed? how is any of this reasonable?

**Comment by raphaelrb96** [comment]
> Eu resolvo meu problema com
> 
> npx react-native run-android --terminal powershell
> 
> no entanto, quero usar apenas npx react-native run-Android

funcionou!!

**Comment by abid-2362** [comment]
on MacOS M1 14.2.1 (23C71)
```
react-native run-android --terminal terminal
```
worked for me, and then for easy access, I added this under the `scripts` in package.json file

```
"android": "react-native run-android --terminal terminal",
"ios": "react-native run-ios --terminal terminal",
```
and then `yarn android` works fine

**Comment by RobertSavage** [comment]
**Running**: `npx react-native run-ios` or `run ios` or `react-native run-ios --terminal terminal`
**Error**: EACCES: permission denied

**Running**: `sudo npx react-native run-ios` or `sudo run ios` or `sudo react-native run-ios --terminal terminal`
**Error**: Cannot start server in new window because no terminal app was specified.

**Fix**: As suggested above I ran `npm run start ` with my simulator open. It did not start the app up for me like it usually does but I was able to click into the app and it was running.

MacOS Sonoma 14.1
Apple M3 Pro
